packer/vhd-image-builder.json

```diff
@@ -194,7 +194,9 @@
             "inline": [
                 "sudo FEATURE_FLAGS={{user `feature_flags`}} BUILD_NUMBER={{user `build_number`}} BUILD_ID={{user `build_id`}} COMMIT={{user `commit`}} /bin/bash -ux /home/packer/install-dependencies.sh",
                 "sudo /bin/bash -eux /home/packer/cleanup-vhd.sh",
-                "sudo rm -rf /home/packer"
+                "sudo rm -rf /home/packer",
+                "cloud-init clean --logs",
```

Comment:
@CecileRobertMichon Is this what you're thinking?

Reply:
yes. Is the diff with the Packer official examples intentional?  https://github.com/hashicorp/packer/blob/master/examples/azure/ubuntu_quickstart.json

compare `/usr/sbin/waagent -force -deprovision+user && export HISTSIZE=0 && sync` with what you have below: `waagent -force -deprovision`

Reply:
Validated that the longer version works

```
azureuser@k8s-master-99506793-0:~$ sudo /usr/sbin/waagent -force -deprovision+user && export HISTSIZE=0 && sync
WARNING! The waagent service will be stopped.
WARNING! Cached DHCP leases will be deleted.
WARNING! root password will be disabled. You will not be able to login as root.
WARNING! /etc/resolvconf/resolv.conf.d/tail and /etc/resolvconf/resolv.conf.d/original will be deleted.
WARNING! azureuser account and entire home directory will be deleted.
azureuser@k8s-master-99506793-0:~$ echo $?
0
```

Reply:
Actually do we want to use the "+user" option to get rid of user info?

Reply:
yes but in that case the user should be packer not azureuser

Reply:
see my comment on the other line

Reply:
It's not clear how to tell waagent which users you want to "deprovision"

Reply:
you don't, it's just the user that the process is running from - in that case it will be packer